presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDataSizeFunctions.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.type.SqlDecimal;
+import com.facebook.presto.spi.type.Type;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.facebook.presto.spi.type.DecimalType.createDecimalType;
+
+public class TestDataSizeFunctions
+        extends AbstractTestFunctions
+{
+    private static final Type DECIMAL = createDecimalType(38, 0);
+
+    @Test
+    public void testParseDataSize()
+    {
+        assertFunction("parse_data_size('0B')", DECIMAL, decimal(0));
+        assertFunction("parse_data_size('1B')", DECIMAL, decimal(1));
+        assertFunction("parse_data_size('2.2kB')", DECIMAL, decimal(2.2 * 1024.0));
+        assertFunction("parse_data_size('2.23kB')", DECIMAL, decimal(2.23 * 1024.0));
+        assertFunction("parse_data_size('2.234kB')", DECIMAL, decimal(2.234 * 1024.0));
+        assertFunction("parse_data_size('3MB')", DECIMAL, decimal(3.0 * 1024.0 * 1024.0));
```

Comment:
Why does this use doubles for computing 3 * 1024 * 1024?

Reply:
It uses `double` to mimic the precision loose introduced by the DataSize parsing. For this particular case it probably doesn't matter. I aggree with @electrum , i think we should re-implement the parsing logic using `BigDecimal`.